src/test/java/uk/gov/pay/directdebit/mandate/resources/MandateResourceIT.java

```diff
@@ -87,70 +93,56 @@ public void tearDown() {
     private GatewayAccountFixture gatewayAccountFixture = GatewayAccountFixture.aGatewayAccountFixture();
 
     private PayerFixture payerFixture = PayerFixture.aPayerFixture();
+    
+    private static ObjectMapper objectMapper = new ObjectMapper();
 
     @Before
     public void setUp() {
         wireMockAdminUsers.start();
         gatewayAccountFixture.insert(testContext.getJdbi());
     }
-
+    
     @Test
-    public void shouldCreateAMandateWithoutPayment_IfMandateIsOnDemand() throws Exception {
+    @Parameters({
+            "null, test-service-ref, Field [return_url] cannot be null",
+            " , test-service-ref, Field [return_url] must have a size between 1 and 255",
+            "http://example, null, Field [service_reference] cannot be null",
+            "http://example, , Field [service_reference] must have a size between 1 and 255"
+    })
+    public void createMandateValidationFailures(@Nullable String returnUrl, 
+                                                @Nullable String serviceReference, 
+                                                String expectedErrorMessage) throws Exception {
         String accountExternalId = gatewayAccountFixture.getExternalId();
-        String returnUrl = "http://example.com/success-page/";
-        String postBody = new ObjectMapper().writeValueAsString(ImmutableMap.builder()
-                .put("return_url", returnUrl)
-                .build());
-
-        String requestPath = "/v1/api/accounts/{accountId}/mandates"
-                .replace("{accountId}", accountExternalId);
+        
+        Map<String, String> createMandateRequest = new HashMap<>();
+        Optional.ofNullable(returnUrl).ifPresent(x -> createMandateRequest.put("return_url", x));
+        Optional.ofNullable(serviceReference).ifPresent(x -> createMandateRequest.put("service_reference", x));
 
-        ValidatableResponse response = givenSetup()
-                .body(postBody)
-                .post(requestPath)
+        givenSetup()
+                .body(objectMapper.writeValueAsString(createMandateRequest))
+                .post("/v1/api/accounts/{accountId}/mandates".replace("{accountId}", accountExternalId))
```

Comment:
Out of curiosity is there a preference to using `.replace` over `format("/v1/api/accounts/%s/mandates", accountExternalId)`?

Reply:
nope. i just copied it from something similar that was there in the test already.